apps/speed.c

```diff
@@ -2800,7 +2800,9 @@ int speed_main(int argc, char **argv)
         for (i = 0; i < loopargs_len; i++)
             ASYNC_WAIT_CTX_free(loopargs[i].wait_ctx);
 
-        ASYNC_cleanup_thread();
+        if (loopargs_len != 0) {
```

Comment:
I think it would be better to create a whole new variable for this rather than attempt to reuse an existing one that is set for a different purpose. The reason is it would be easy for someone to move around the relative ordering of where `loopargs_len` is set and `ASYNC_init_thread()` is called, or perhaps, add new code between the two statements which adds a new `goto end`.

It would be even nicer if ASYNC_cleanup_thread() was a no-op if ASYNC_init_thread() hasn't been called yet...but that might be too hard.